File: modules/@angular/core/testing/lang_utils.dart
library testing.lang_utils;

import 'dart:mirrors';

Type getTypeOf(instance) => instance.runtimeType;

dynamic instantiateType(Type type, [List params = const []]) {
  var cm = reflectClass(type);
  return cm.newInstance(new Symbol(''), params).reflectee;
}

File: modules/@angular/core/testing/lang_utils.ts
export function getTypeOf(instance: any /** TODO #9100 */) {
  return instance.constructor;
}

export function instantiateType(type: Function, params: any[] = []) {
  var instance = Object.create(type.prototype);
  instance.constructor.apply(instance, params);
  return instance;
}
